admin-dev/themes/new-theme/js/pages/order/view/order-discounts-refresher.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import Router from '@components/router';
+import OrderViewPageMap from '@pages/order/OrderViewPageMap';
+
+const $ = window.$;
+
+export default class OrderDiscountsRefresher {
+  constructor() {
+    this.router = new Router();
+  }
+
+  refresh(orderId) {
+    $.ajax(this.router.generate('admin_orders_get_discounts', {orderId}))
+      .then((response) => {
+        $(OrderViewPageMap.productDiscountList.list).replaceWith(response);
```

Comment:
Do you think it would be complicated to return a JSON instead in the controller? And manage the update via javascript?
Instead of adding a new ajax controller that returns some HTML code

Reply:
It would be a big change of scope for this PR.
This requires heavy javascript logic and it means that modules that would customize/override the Twig template will not see their changes being kept after ajax refresh (this is why we dont vue.js yet in BO). I suggest we stick to HTML replace 😉

Reply:
The problem is that (if we do it) we need to do it before beta release, for not changing the signature of the API.

Reply:
Not an issue. If we want to modify it later, we can create a new API endpoint and deprecate the precious one. Let's continue this on slack 😉